### DISCUSSION TITLE
Go2RTC setup with Homeassistant on v0.12

### DISCUSSION THREAD

**Discussion by hagak** [blakeblackshear/frigate#6123] [Ask A Question]
So I recently upgraded to v0.12 and moved my cameras over to a go2rtc restreams.  I have tested connecting to the cameras via WebRTC within the Frigate UI and that works great.

However my issue comes up when I want to connect HA to Frigate using webrtc.  HA with the Frigate interface 4.0 is installed and can view the camera but it is not using webrtc.  From what I understand I need to install Alexxx's WebRTC Camera integration and point it to my frigate ip on port 1984.

Question I have is I have frigate running as a container alongside HA.  Can I tell HA to connect to frigate within docker over 1984, ie frigate_docker_ip:1984 vs opening a port to frigate on 1984 and use the host_ip:1984.  Second is HA the WebRTC client or is the user browser going to HA the WebRTC client?

**Comment by NickM-27** [comment]
I'd suggest opening port 1984 and then using host ip. 

That being said, version 5 of the HA card also supports WebRTC and mse, though it is still in beta.

**Reply by hagak** [reply]
If HA is the only thing that communicates to Frigate of 1984 then is it really needed to open the port.

I admit I am not versed at all how WebRTC functions.  If I connect webrtc camera to host_ip:1984 or frigate_docker_ip:1984 I get the same result.  The integration adds successfully but if I try to select WebRTC Camera as a Live view Provider the browser spins and becomes unresponsive. And HA outputs a ton of errors repeating on:
```
Traceback (most recent call last):
  File "/config/custom_components/frigate/media_source.py", line 684, in async_browse_media
    events = await self._get_client(identifier).async_get_events(
  File "/config/custom_components/frigate/api.py", line 85, in async_get_events
    await self.api_wrapper(
  File "/config/custom_components/frigate/api.py", line 217, in api_wrapper
    raise FrigateApiClientError from exc
custom_components.frigate.api.FrigateApiClientError

The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 26, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/media_source/__init__.py", line 170, in websocket_browse_media
    media = await async_browse_media(hass, msg.get("media_content_id", ""))
  File "/usr/src/homeassistant/homeassistant/components/media_source/__init__.py", line 121, in async_browse_media
    item = await _get_media_item(hass, media_content_id, None).async_browse()
  File "/usr/src/homeassistant/homeassistant/components/media_source/models.py", line 79, in async_browse
    return await self.async_media_source().async_browse_media(self)
  File "/config/custom_components/frigate/media_source.py", line 695, in async_browse_media
    raise MediaSourceError from exc
```

**Reply by NickM-27** [reply]
these logs have to do with the frigate integration which is entirely unrelated to the webrtc camera integration. 

> If HA is the only thing that communicates to Frigate of 1984 then is it really needed to open the port.

well, if you are going to use the docker IP then you better make sure you have it setup as static. By default docker IPs change when the host restarts so your setup will break on each restart

**Reply by hagak** [reply]
> these logs have to do with the frigate integration which is entirely unrelated to the webrtc camera integration.
> 
> > If HA is the only thing that communicates to Frigate of 1984 then is it really needed to open the port.
> 
> well, if you are going to use the docker IP then you better make sure you have it setup as static. By default docker IPs change when the host restarts so your setup will break on each restart

Well you of course don't use the docker ip you define a container name and use that.  I do this for many containers that are just supporting container for others (mysql, etc).  This works perfectly well AS long as the client is also on the same docker network.  So if HA is the client in the comms to Frigate on 1984 then that should be fine.

**Reply by NickM-27** [reply]
Sure, that wasn't what was described above and many users make the mistake of using the direct docker IP so wanted to make sure that was not done. 

In any case, when you try to set the webrtc card as the live view provider you need to give it other information as well like a url: for the camera. You also need to be running v5 of the frigate card which I'd suggest just using the built in go2rtc live provider instead

**Reply by hagak** [reply]
No problem, good to be clear and I was not.

Just installed v5 and very nice, I just had to edit the card for each camera to use go2rtc and it just WORKED! 

So currently I do not have port 8555 mapped into my network from external, I expected it to just fail but it looks like it might fall back to a non-webrtc protocol?  Is that the case?

**Reply by NickM-27** [reply]
Yes it will use mse if WebRTC can't be resolved.
